jmclient/jmclient/taker.py

```diff
@@ -820,22 +828,44 @@ def self_sign_and_push(self):
         self.self_sign()
         return self.push()
 
+    def tx_match(self, txd):
+        # Takers process only in series, so this should not occur:
+        assert self.latest_tx is not None
+        # check if the transaction matches our created tx:
+        if txd['outs'] != self.latest_tx['outs']:
+            return False
+        return True
+
     def unconfirm_callback(self, txd, txid):
+        if not self.tx_match(txd):
```

Comment:
under what circumstances can this happen?

Reply:
Same as above discussion, each tx fires the extant callbacks, so we check if it matches. In Taker this must be exactly `self.latest_tx` not a dict since Taker processes only in series.